backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/EnumClassLowering.kt

```diff
@@ -124,19 +125,25 @@ internal class EnumUsageLowering(val context: Context)
     }
 }
 
-internal class EnumClassLowering(val context: Context) : ClassLoweringPass {
+internal class EnumClassLowering(val context: Context) : FileLoweringPass {
 
     fun run(irFile: IrFile) {
         // EnumWhenLowering should be performed before EnumUsageLowering because
         // the latter performs lowering of IrGetEnumValue
         EnumWhenLowering(context).lower(irFile)
-        runOnFilePostfix(irFile)
+        lower(irFile)
         EnumUsageLowering(context).lower(irFile)
     }
 
-    override fun lower(irClass: IrClass) {
-        if (irClass.kind != ClassKind.ENUM_CLASS) return
-        EnumClassTransformer(irClass).run()
+    override fun lower(irFile: IrFile) {
+        irFile.transformChildrenVoid(object : IrElementTransformerVoid() {
+            override fun visitClass(declaration: IrClass): IrStatement {
+                declaration.transformChildrenVoid()
+                if (declaration.kind == ClassKind.ENUM_CLASS)
+                    EnumClassTransformer(declaration).run()
+                return declaration
+            }
```

Comment:
Sorry I didn't get the difference. Why is FileLowering better?

Reply:
No difference, indeed, just seemed a bit more logical since we are gettting an `irFile` as input.